ee/debug/checkups/download_directory.go

```diff
@@ -0,0 +1,119 @@
+package checkups
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+type downloadDirectory struct {
+	status  Status
+	summary string
+	files   []fileInfo
+}
+
+func (c *downloadDirectory) Name() string {
+	return "Download directory contents for all users"
+}
+
+func (c *downloadDirectory) Run(_ context.Context, extraFH io.Writer) error {
+	downloadDirs := getDownloadDirs()
+	if len(downloadDirs) == 0 {
+		c.status = Erroring
+		c.summary = "No download directories found"
+		return nil
+	}
+
+	for _, downloadDir := range downloadDirs {
+		pattern := filepath.Join(downloadDir, "kolide-launcher*")
+		matches, err := filepath.Glob(pattern)
+
+		if err != nil {
+			fmt.Fprintf(extraFH, "Error listing files in directory (%s): %s\n", downloadDir, err)
+			continue
+		}
+
+		for _, match := range matches {
+			if info, err := os.Stat(match); err == nil {
+				c.files = append(c.files, fileInfo{
+					Name:    match,
+					ModTime: info.ModTime(),
+				})
+			}
+		}
+	}
+
+	switch {
+	case len(c.files) == 0:
+		c.status = Informational
+		c.summary = "No Kolide installers found in any user's download directory"
```

Comment:
FWIW `status` and `summary` mostly matter for doctor commands, I basically always ignore them for flares